tools/update-test-manifests

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python3
+"""
+Updates the repository snapshot date defined in the 'vars' fragments in test/data/manifests/ and regenerates all test
+manifests.
+"""
+import json
+import os
+import pathlib
+import subprocess as sp
+import sys
+from glob import glob
+from tempfile import TemporaryDirectory
+
+from ruamel.yaml import YAML
+
+
+def read_schutzfile():
+    with open("Schutzfile", mode="r", encoding="utf-8") as schutzfile:
+        return json.load(schutzfile)
+
+
+def find_snapshot_date(distro, arch, schutzfile):
+    # Each distro can have multiple lists of repositories. We assume all share the same snapshot date, so we only check
+    # the first one.
+    repos = schutzfile[distro]["repos"][0][arch]
+    first_url = repos[0]["baseurl"]
+    *_, date = first_url.split("-")
+    return date
+
+
+def main():
+    if os.getuid() != 0:
+        print("This script requires root to build manifests", file=sys.stderr)
+        sys.exit(1)
+
+    vars_files = glob("test/data/manifests/*-vars.ipp.yaml")
```

Comment:
If you replace this with `Path("text/data/manifests/").glob("*-vars.ipp.yaml")` you directly get an iterable of Path objects and don't need to convert again on line 45.